src/Surfnet/StepupSelfService/SelfServiceBundle/Service/ActivationFlowService.php

```diff
@@ -71,38 +113,52 @@ public function process(string $uri): void
                     $uri
                 )
             );
-            return;
+            return new ActivationFlowPreferenceNotExpressed();
         }
 
         $option = $parameters[$this->fieldName];
-        if (!in_array($option, $this->options)) {
-            $this->logger->notice(
-                sprintf(
-                    'Field "%s" contained an invalid option "%s", must be one of: %s',
-                    $this->fieldName,
-                    $option,
-                    implode(', ', $this->options)
-                )
-            );
-            return;
+        switch ($option) {
+            case 'ra':
+                return ActivationFlowPreference::createRa();
+            case 'self':
+                return ActivationFlowPreference::createSelf();
```

Comment:
Match against the allowed/configured $this->options? The array keys of that array contain the `ra` and `self` options. I see this is a very marginal gain feel free to ignore